addons/mail/models/mail_notification.py

```diff
@@ -57,7 +58,8 @@ class MailNotification(models.Model):
     def init(self):
         self._cr.execute("""
             CREATE INDEX IF NOT EXISTS mail_notification_res_partner_id_is_read_notification_status_mail_message_id
-                                    ON mail_notification (res_partner_id, is_read, notification_status, mail_message_id)
+                                    ON mail_notification (res_partner_id, is_read, notification_status, mail_message_id);
+            CREATE INDEX IF NOT EXISTS mail_notification_author_id_notification_status_failure ON mail_notification (author_id, notification_status) WHERE notification_status in ('bounce', 'exception');
```

Comment:
Linting: could you multi-line it :) ? Going outside of my screen ^^ .

Reply:
Sure, it's fine for me either way.